packages/auth-core/src/core.ts

```diff
@@ -175,39 +175,27 @@ export class Auth {
 
   async getProvidersInfo() {
     // TODO: cache this data when we have a way to invalidate on config update
-    let providers: {
-      _typename: string;
-      name: string;
-      display_name: string | null;
-    }[];
     try {
-      providers = await this.client.query(`
-      with module ext::auth
-      select cfg::Config.extensions[is AuthConfig].providers {
-        _typename := .__type__.name,
-        name,
-        [is OAuthProviderConfig].display_name,
+      return await this.client.queryRequiredSingle<{
+        oauth: { name: string; display_name: string }[];
+        emailPassword: boolean;
+      }>(`
+      with
+        module ext::auth,
+        providers := (select cfg::Config.extensions[is AuthConfig].providers)
+      select {
+        oauth := providers[is OAuthProviderConfig] {
+          name,
+          display_name
+        },
+        emailPassword := exists providers[is EmailPasswordProviderConfig]
```

Comment:
Not a blocker: it seems like we should still maybe return the `name` here so we don't have a magic const string in the code?

Reply:
I'm not sure that would have much use, since I decided to not expose that detail elsewhere as the api's are specific to the email/password provider anyway. But internally, yes, it's probably a good idea to use the emailPasswordProvider name from the generated consts file, instead of how it's currently hardcoded.